python/ray/train/impl/data_parallel_trainer.py

```diff
@@ -0,0 +1,217 @@
+import logging
+import os
+from typing import Dict, Callable, Optional, Type, TYPE_CHECKING
+
+import ray
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.config import ScalingConfig, RunConfig, _DataParallelScalingConfig
```

Comment:
Should probably make DataParallelScalingConfig public if most trainers are going to use it.